### ISSUE TITLE
Top-level await is currently not supported with the "cjs" output format

### ISSUE THREAD

**Issue by astoilkov** [privatenumber/tsx#488]
### Acknowledgements

- [X] I searched existing issues before opening this one to avoid duplicates
- [X] I understand this is not a place for seek help, but to report a bug
- [X] I understand that the bug must be proven first with a minimal reproduction
- [X] I will be polite, respectful, and considerate of people's time and effort

### Minimal reproduction URL

https://stackblitz.com/edit/node-mkxshw?file=tsconfig.json,index.ts

### Version

v4.7.1

### Node.js version

18.18.2

### Package manager

yarn

### Operating system

macOS

### Problem & Expected behavior

Running `tsx script.ts` throws `Top-level await is currently not supported with the "cjs" output format` error. I expected the file to run.

### Contributions

- [ ] I plan to open a pull request for this issue
- [ ] I plan to make a financial contribution to this project

**Comment by tweeres04** [comment]
tldr: add `"type": "module"` to your `package.json`
